hbase-server/src/main/java/org/apache/hadoop/hbase/master/HMaster.java

```diff
@@ -378,12 +381,18 @@ public class HMaster extends HBaseServerBase<MasterRpcServices> implements Maste
 
   private HbckChore hbckChore;
   CatalogJanitor catalogJanitorChore;
-  // Threadpool for scanning the archive directory, used by the HFileCleaner
-  private DirScanPool hfileCleanerPool;
   // Threadpool for scanning the Old logs directory, used by the LogCleaner
   private DirScanPool logCleanerPool;
   private LogCleaner logCleaner;
-  private HFileCleaner hfileCleaner;
+  // HFile cleaners for the custom hfile archive paths and the default archive path
+  // The archive path cleaner is the first element
+  private LinkedList<HFileCleaner> hfileCleaners = new LinkedList<>();
```

Comment:
The choice of LinkedList will trigger some static analysis tools to warn it is almost always a poor choice. In my opinion the use of LinkedList here makes sense but ArrayList would avoid that issue. Just a suggestion.

Reply:
Thanks, good suggestion!